inference-engine/src/cldnn_engine/cldnn_program.h

```diff
@@ -126,7 +127,10 @@ class Program {
 
     template<typename OpType, typename std::enable_if<std::is_base_of<ngraph::Node, OpType>::value, int>::type = 0>
     static void RegisterFactory(factory_t func) {
-        Program::factories_map.insert({OpType::type_info, func});
+        static std::mutex m;
```

Comment:
looks like in CPU plugin we don't have mutex. What is the difference?

Reply:
Looks like CPU plugin creates unique `MKLDNNExtensions` object for each plugin instance, so there's no data races.